### DISCUSSION TITLE
Error: Failed to collect page data for /api/auth/[...nextauth]

### DISCUSSION THREAD

**Discussion by hamadodene** [vercel/community#4942] [Help] [answered]
### Summary

Hi,
i'm trying do deploy my Next13 web app to vercel but i always get this error during build process. 

```
Learn how: https://pris.ly/d/vercel-build
13:30:56.157 | at Ws (/vercel/path0/eduwise/node_modules/@prisma/client/runtime/library.js:114:69)
13:30:56.157 | at new t (/vercel/path0/eduwise/node_modules/@prisma/client/runtime/library.js:126:2533)
13:30:56.157 | at 86821 (/vercel/path0/eduwise/.next/server/chunks/783.js:106:37)
13:30:56.158 | at __webpack_require__ (/vercel/path0/eduwise/.next/server/webpack-runtime.js:25:43)
13:30:56.158 | at 19212 (/vercel/path0/eduwise/.next/server/chunks/783.js:16:71)
13:30:56.158 | at __webpack_require__ (/vercel/path0/eduwise/.next/server/webpack-runtime.js:25:43)
13:30:56.158 | at 33206 (/vercel/path0/eduwise/.next/server/app/api/auth/[...nextauth]/route.js:130:19)
13:30:56.158 | at __webpack_require__ (/vercel/path0/eduwise/.next/server/webpack-runtime.js:25:43)
13:30:56.159 | at __webpack_exec__ (/vercel/path0/eduwise/.next/server/app/api/auth/[...nextauth]/route.js:177:39)
13:30:56.159 | at /vercel/path0/eduwise/.next/server/app/api/auth/[...nextauth]/route.js:178:81 {
13:30:56.159 | clientVersion: '5.1.1',
13:30:56.159 | errorCode: undefined
13:30:56.159 | }
13:30:56.165 |  
13:30:56.166 | > Build error occurred
13:30:56.168 | Error: Failed to collect page data for /api/auth/[...nextauth]
13:30:56.168 | at /vercel/path0/eduwise/node_modules/next/dist/build/utils.js:1156:15
13:30:56.168 | at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
13:30:56.168 | type: 'Error'
13:30:56.168 | }
13:30:56.242 | Error: Command "npm run build" exited with 1
13:30:56.647

```

### Steps to Reproduce

- I setup  nextauth using next 13 app directory following this step: https://next-auth.js.org/configuration/initialization#route-handlers-app

**Comment by TobiasGleiter** [comment]
Did you find a solution?

**Reply by jiaegong** [reply]
Did you check your env file? I had same issue and I just copied my key file and it made new line so couldn't read private key collectly. Check your private key again!

**Comment by MiladUchiha** [comment]
Hi Im facing same issue
Can anyone help?

**Reply by jiaegong** [reply]
Did you check your env file? I had same issue and I just copied my key file and it made new line so couldn't read private key collectly. Check your private key again!

**Comment by RadiBarq** [comment, marked as answer]
I had this issue when the NEXTAUTH_URL="" was missing from the .env file it's fixed when added.

**Reply by amirrezabahramy** [reply]
This solution worked. Thanks.

**Reply by tgoulder4** [reply]
This solution did not work for me.

**Reply by prashant-2204** [reply]
still not wroking for me
I am using v5

**Reply by amitamrutiya** [reply]
Same issue in v5

**Reply by tushcmd** [reply]
same issue here

**Reply by NikosCha** [reply]
USE AUTH_URL instead of NEXTAUTH_URL

**Reply by prashant-2204** [reply]
use prisma as adaptor to connect with mongoDB..it will work fine

**Comment by tushcmd** [comment]
Make sure you have run `prisma generate ` before deploying. I added 
```json
"scripts": {
  "postinstall": "prisma generate"
}
```
to package.json

**Reply by kk7188048** [reply]
Still its not working, Can anyone help?

**Reply by amitamrutiya** [reply]
Yeah, It's not working

**Reply by naji-alhusami** [reply]
thanks, this worked with me:
"scripts": {
  "postinstall": "prisma generate"
}

and I run npm install after I added it
